vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpacesCustomOrderManager.java

```diff
@@ -0,0 +1,152 @@
+import edu.asu.diging.vspace.core.model.IExhibition;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.ISpacesCustomOrder;
+import edu.asu.diging.vspace.core.model.impl.Exhibition;
+import edu.asu.diging.vspace.core.model.impl.SpaceStatus;
+import edu.asu.diging.vspace.core.model.impl.SpacesCustomOrder;
+import edu.asu.diging.vspace.core.services.IExhibitionManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+import edu.asu.diging.vspace.core.services.ISpacesCustomOrderManager;
+
+/**
+ * SpacesCustomOrderManager is the manager
+ * to allow custom ordering of spaces
+ * @author Glen D'souza
+ *
+ */
+
+@Service
+@Transactional(rollbackFor = { Exception.class })
+public class SpacesCustomOrderManager implements ISpacesCustomOrderManager {
+    
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private SpacesCustomOrderRepository spacesCustomOrderRepository;
+    
+    @Autowired
+    private ISpaceManager spaceManager;
+    
+    @Autowired
+    private IExhibitionManager exhibitionManager;
+    
+    @Override
+    public ISpacesCustomOrder create(List<String> spaceOrders,
+            String name,
+            String description) {
+        SpacesCustomOrder spacesCustomOrder = new SpacesCustomOrder();
+        spacesCustomOrder.setCustomOrderName(name);
+        spacesCustomOrder.setDescription(description);
+        List<ISpace> orderedSpaces = spaceOrders.stream()
+                .map(spaceId -> spaceManager.getSpace(spaceId))
+                .collect(Collectors.toList());
+        spacesCustomOrder.setCustomOrderedSpaces(orderedSpaces);
+        return spacesCustomOrderRepository.save(spacesCustomOrder);
+    }
+
+    
+    @Override
+    public Iterable<SpacesCustomOrder> findAll(){
+        return spacesCustomOrderRepository.findAll();
+    }
+    
+    @Override
+    public ISpacesCustomOrder get(String customSpaceOrderId) {
+        Optional<SpacesCustomOrder> spacesCustomOrderOptional = spacesCustomOrderRepository.findById(customSpaceOrderId);
+        if(spacesCustomOrderOptional.isPresent())
+            return spacesCustomOrderOptional.get();
+        return null;
+    }
+    
+    @Override
+    public void save(Iterable<SpacesCustomOrder> spacesCustomOrder) {
+        spacesCustomOrderRepository.saveAll(spacesCustomOrder);
+    }
+
+    /**
+     * This method adds the newly created spaces to all custom orders at the end of the list
+     * @param space
+     */
+    @Override
+    public void addSpaceToCustomOrders(ISpace space) {
+        Iterable<SpacesCustomOrder> spacesCustomOrders = findAll();
+        for(ISpacesCustomOrder spaceCustomOrder :  spacesCustomOrders) {
+            spaceCustomOrder.getCustomOrderedSpaces().add(space);
+        }
+        save(spacesCustomOrders);
+    }
+    
+    /**
+     *This method updates the custom order name or description
+     */
```

Comment:
needs complete javadoc (params are missing). also, this does not update name or description, it updates both